lib/frames.js

```diff
@@ -1,16 +1,32 @@
 'use strict';
 var Builder = require('buffer-builder'),
-    AMQPFrame = require('./frames/frame').AMQPFrame,
-    Frame = require('./frames/frame').Frame,
-    TransferFrame = require('./frames/transfer_frame'),
+    DescribedType = require('./types/described_type'),
     errors = require('./errors'),
     codec = require('./codec'),
-    debug = require('debug')('amqp10:framing');
+    debug = require('debug')('amqp10:framing'),
+    constants = require('./constants'),
+    types = require('./types'),
+    terminus = require('./types/terminus'),
+    util = require('util');
 
 var frames = module.exports = {};
+var framesByDescriptor = {};
+
+var FrameType = { AMQP: 0x00, SASL: 0x01 };
+frames.Frame = function(type) { this.type = type; };
+frames.AMQPFrame = function(channel) {
+  frames.AMQPFrame.super_.call(this, constants.frameType.amqp);
+  this.channel = channel || 0;
+};
+util.inherits(frames.AMQPFrame, frames.Frame);
+
+frames.SaslFrame = function() {
+  frames.SaslFrame.super_.call(this, constants.frameType.sasl);
+};
+util.inherits(frames.SaslFrame, frames.Frame);
 
 frames.writeFrame = function(frame, stream) {
-  if (!(frame instanceof Frame)) {
+  if (!(frame instanceof frames.Frame)) {
     throw new errors.EncodingError('unknown frame type: ' + frame.type);
```

Comment:
honestly, might make sense to use JSON.stringify instead of `frame.type` here, since we already know it's not a Frame so probably doesn't have a type (or if it does, it's accidental).